client/components/app.jsx

```diff
@@ -48,12 +53,32 @@ class App extends React.Component {
       });
   }
 
+  deleteGrades(Id) {
+    let eventTarget = null;
+    const grades = this.state.grades.slice();
+    for (let i = 0; i < grades.length; i++) {
+      if (Id === grades[i].id) {
+        eventTarget = grades[i];
+        break;
+      }
+    }
+    fetch(`/api/grades/${Id}`, {
+      method: 'DELETE'
+    })
+      .then(res => res.json())
+      .then(rs => {
+        grades.slice(eventTarget, 1);
+        this.setState({ grades });
```

Comment:
I recommend using `Array.prototype.filter` to exclude all grades where their `id` is not equal to the `Id` parameter.